### ISSUE TITLE
[$1000] Contact method - Account is locked when invalid email is used as secondary contact method

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#25711]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify,com
2. Go to Settings > Profile > Contact method > New contact method
3. Add an invalid email address (eg, [dusjosjkposakpa@gmail.com](mailto:dusjosjkposakpa@gmail.com)
4. Save and log out
5. Log in with primary contact method

## Expected Result:
User is able to log in without issue

## Actual Result:
The account is locked and user is unable to login with primary contact method

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.56-7

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/e61b3316-82de-4421-8cb4-d4e6e7a01c94

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0174bb34de68da9dd3</li>
        <li>Upwork Job ID: 1694782852370354176</li>
        <li>Last Price Increase: 2023-08-24</li>
    </ul>
</details>

**Comment by MitchExpensify** [comment]
I cannot reproduce this

**Comment by MitchExpensify** [comment]
@lanitochka17 Are you

**Comment by MitchExpensify** [comment]
Reproduced! 
<img width="453" alt="image" src="https://github.com/Expensify/App/assets/36425901/68d1a55e-4833-4374-8a2c-f37dc523efa9">

**Comment by b4s36t4** [comment]
<img width="415" alt="Screenshot 2023-08-25 at 12 32 04 AM" src="https://github.com/Expensify/App/assets/59088937/a23fd079-8223-4756-b26c-4452ffc7d737">

I believe this is an issue with backend. `hasEmailDeliveryFailure` field is true which is reason to show the above error page.

**Comment by MitchExpensify** [comment]
Labelling internal as this appears to be a backend issue

**Comment by uniuuu** [comment]
> <img alt="Screenshot 2023-08-25 at 12 32 04 AM" width="415" src="https://user-images.githubusercontent.com/59088937/263077609-a23fd079-8223-4756-b26c-4452ffc7d737.png">
> 
> I believe this is an issue with backend. `hasEmailDeliveryFailure` field is true which is reason to show the above error page.

Regarding where and why `hasEmailDeliveryFailure` boolean used can be seen in next function [getRenderOptions](https://github.com/Expensify/App/blob/c16995257c738761206ea07645e863ee717ecbc0/src/pages/signin/SignInPage.js#L76) . 

This issue is due to missing check for email delivery issues at the moment the invalid email added to secondary contact method. The best way to fix it is to validate email after it's key in when adding it. And do not allow to save an invalid email.
Otherwise if invalid email will be allowed to be save and propose changes for `getRenderOptions` to let it pass sign-in in case primary method is valid then such approach will lead to failure with different scenarios (i.e. in case user can't use primary method) and therefore it's not good.
In this point of view it's not bug but missing feature.

**Comment by uniuuu** [comment]
Contributor details
Your Expensify account email: alex.kred@yahoo.co.uk
Upwork Profile Link: https://www.upwork.com/freelancers/oleksandrk89

**Comment by mollfpr** [comment]
@uniuuu I believe we already have the response error for the invalid email submitted to notify that the backend can't send any email confirmation.

The email used in the issue description seems normal. So I'm not sure what we can do in the frontend to prevent this.

**Comment by mollfpr** [comment]
@MitchExpensify I think we need to know what the criteria are to lock the account when adding the secondary contact method. Could we pull the internal engineer to check it?

**Comment by MitchExpensify** [comment]
Applying Eng label for help with this:

> I think we need to know what the criteria are to lock the account when adding the secondary contact method. (https://github.com/Expensify/App/issues/25711#issuecomment-1705571304)

**Comment by dangrous** [comment]
Reading through this, let me see what I can find out! Does feel like a backend issue, certainly.

I feel like I saw in Slack this happening to some contributors previously. I also feel like we recently changed something with this due to something happening with secondary logins that are phone numbers. That is all very vague (and may be incorrect haha) but maybe it will spark something while I'm looking
